includes/class-parcelpro-api.php

```diff
@@ -305,4 +305,84 @@ public function types_key_values()
 
         return(($result_records));
     }
+
+    /**
+     * @param string $carrier The carrier name.
+     * @param DateTimeInterface $dateTime The date on which the package will be handed over to the carrier.
+     * @param $postcode string The postal code of the package destination.
+     *
+     * @return DateTimeImmutable|false
+     */
+    public function getDeliveryDate(string $carrier, \DateTimeInterface $dateTime, string $postcode)
+    {
+        if (!$postcode) {
+            return $postcode;
+        }
+        $postcode = str_replace(' ', '', $postcode);
+
+        $date = $dateTime->format('Y-m-d');
+
+        $query = http_build_query([
+            'Startdatum' => $date,
+            'Postcode' => $postcode,
+            'GebruikerId' => $this->login_id,
+            'Map' => true,
+        ]);
+
+        $curlHandle = curl_init();
+        curl_setopt_array($curlHandle, [
+            CURLOPT_URL => $this->api_url . '/api/v3/timeframes.php?' . $query,
+            CURLOPT_HTTPHEADER => [
+                'Content-Type: application/json',
+                'Digest: ' . hash_hmac(
+                    "sha256",
+                    sprintf('GebruikerId=%sPostcode=%sStartdatum=%s', $this->login_id, $postcode, $date),
+                    $this->api_id
+                ),
+            ],
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_HEADER => false,
+            CURLOPT_RETURNTRANSFER => true,
+        ]);
+
+        $responseBody = curl_exec($curlHandle);
+        $responseCode = curl_getinfo($curlHandle, CURLINFO_RESPONSE_CODE);
+
+        curl_close($curlHandle);
+
+        if ($responseCode !== 200) {
+            $logger = wc_get_logger();
+            if ($logger) {
+                $logger->error(sprintf(
+                    'Failed to get expected delivery date, response code %s, body:\n%s',
+                    $responseCode,
+                    $responseBody
+                ));
+            }
+            return "200";
+        }
+
+        $responseJson = json_decode($responseBody, true);
+        $rawDate = false;
+
+        // Lookup the expected date in a case-insensitive way, since a carrier may be "Postnl" or PostNL".
```

Comment:
Is dit daadwerkelijk zo? Uit de API weten we precies wat we terugkrijgen, en ik meen dat de carrier namen in de plugin diezelfde casing gebruiken - al weet ik dat niet 100% zeker.

Reply:
Ik hoopte dat dit ook zo was, maar de namen van de plugin kwamen met een andere casing terug al. Heb daarom na een `var_dump` en een poos debuggen wat de bron ervan was, besloten om het zo failsafe te implementeren